gazebo_state_plugins/src/GazeboObjectInfo.cpp

```diff
@@ -116,7 +138,20 @@ shape_msgs::SolidPrimitive * GazeboObjectInfo::getSolidPrimitive(physics::Collis
             ROS_ERROR("Dynamic cast failed for box shape");
             return NULL;
         }
-        math::Vector3 bb=box->GetSize();
+
+#if GAZEBO_MAJOR_VERSION >= 8
+        gz_math_Vector3d bb=box->Size();
+        if ((bb.X() < 1e-05) || (bb.Y() < 1e-05) || (bb.Z() < 1e-05)){
+            ROS_WARN_ONCE("Skipping coll %s because its bounding box is flat",c->GetName().c_str());
+            return NULL;
+        }
+        solid.type=shape_msgs::SolidPrimitive::BOX;
+        solid.dimensions.resize(3);
+        solid.dimensions[shape_msgs::SolidPrimitive::BOX_X]=bb.X();
```

Comment:
Here it would also be nice to minimise code duplication with a function which takes a `GzVector3d` and returns the solid dimensions (the `float64[] dimensions` in `shape_msgs/SolidPrimitive`)